tests/unit/LegalObligation.spec.ts

```diff
@@ -0,0 +1,125 @@
+import Vue from 'vue'
+import Vuetify from 'vuetify'
+import Vuelidate from 'vuelidate'
+import { mount } from '@vue/test-utils'
+import { getVuexStore } from '@/store'
+import flushPromises from 'flush-promises'
+
+// Components
+import LegalObligation from '@/components/Dashboard/LegalObligation.vue'
+
+Vue.use(Vuetify)
+Vue.use(Vuelidate)
+
+const vuetify = new Vuetify({})
+const store = getVuexStore()
+
+const newBusinessFiling = [
+  {
+    'filing': {
+      'header': {
+        'name': 'incorporationApplication',
+        'date': '2019-06-02T19:22:59.003777+00:00',
+        'paymentToken': 123,
+        'certifiedBy': 'Full Name 1',
+        'filingId': 321,
+        'availableOnPaperOnly': false,
+        'effectiveDate': '2019-06-02T19:22:59.003777+00:00'
+      },
+      'incorporationApplication': {
+        'annualGeneralMeetingDate': '2019-12-31',
+        'annualReportDate': '2019-12-31'
+      }
+    }
+  }
+]
+
+const businessWithMaintenanceFiling = [
+  {
+    'filing': {
+      'header': {
+        'name': 'incorporationApplication',
+        'date': '2019-06-02T19:22:59.003777+00:00',
+        'paymentToken': 123,
+        'certifiedBy': 'Full Name 1',
+        'filingId': 321,
+        'availableOnPaperOnly': false,
+        'effectiveDate': '2019-06-02T19:22:59.003777+00:00'
+      },
+      'incorporationApplication': {
+      }
+    }
+  },
+  {
+    'filing': {
+      'header': {
+        'name': 'annualReport',
+        'date': '2019-06-02T19:22:59.003777+00:00',
+        'paymentToken': 123,
+        'certifiedBy': 'Full Name 1',
+        'filingId': 321,
+        'availableOnPaperOnly': false,
+        'effectiveDate': '2019-06-02T19:22:59.003777+00:00'
+      },
+      'annualReport': {
+        'annualGeneralMeetingDate': '2019-12-31',
+        'annualReportDate': '2019-12-31'
+      }
+    }
+  }
+]
+
+async function waitForUpdate () {
```

Comment:
I always enjoy your custom methods inside the unit tests :)

Reply:
Thanks :)